packages/devtools_app/test/logging_controller_test.dart

```diff
@@ -59,36 +69,70 @@ void main() {
       addStdoutData('Abc.');
 
       expect(controller.data, isNotEmpty);
-      expect(controller.filteredData, isNotEmpty);
+      expect(controller.filteredData.value, isNotEmpty);
 
       controller.clear();
 
       expect(controller.data, isEmpty);
-      expect(controller.filteredData, isEmpty);
+      expect(controller.filteredData.value, isEmpty);
+    });
+
+    test('matchesForSearch', () {
+      addStdoutData('abc');
+      addStdoutData('def');
+      addStdoutData('abc ghi');
+      addGcData('gc1');
+      addGcData('gc2');
+
+      expect(controller.filteredData.value, hasLength(5));
+      expect(controller.matchesForSearch('abc').length, equals(2));
+      expect(controller.matchesForSearch('ghi').length, equals(1));
+      expect(controller.matchesForSearch('abcd').length, equals(0));
+      expect(controller.matchesForSearch('').length, equals(0));
+
+      // Search by event kind.
+      expect(controller.matchesForSearch('stdout').length, equals(3));
+      expect(controller.matchesForSearch('gc').length, equals(2));
+
+      // Search with incorrect case.
+      expect(controller.matchesForSearch('STDOUT').length, equals(3));
     });
 
-    test('filteredData', () {
+    test('filterData', () {
       addStdoutData('abc');
       addStdoutData('def');
       addStdoutData('abc ghi');
+      addGcData('gc1');
+      addGcData('gc2');
 
-      expect(controller.data, hasLength(3));
-      expect(controller.filteredData, hasLength(3));
+      expect(controller.data, hasLength(5));
+      expect(controller.filteredData.value, hasLength(5));
 
-      controller.filterText = 'abc';
+      controller.filterData(QueryFilter.parse('abc', controller.filterArgs));
```

Comment:
All the tests for QueryFilter look like they take a single argument. Should we have some tests with multiple arguments? In particular, multiple negative arguments seems relevant.

Reply:
added more tests for logging_controller_test and network_controller_test.